### PR TITLE
Hide password + user form fields when logged in via JWT and SAML

### LINKED ISSUE #23298: [Epic] Remove ability for users to edit SSO created user info
It doesn't make much sense for users who logged in with SSO to be able to change their account information e.g. names, email since their information will already be synced each time users log in (correct me if I'm wrong, but basically SSO data will be syncing to Metabase). Being able to change their own account info makes the data out of sync with SSO providers.

**Links**
- product doc: https://www.notion.so/metabase/Remove-ability-for-users-to-edit-SSO-created-user-info-983a23b1f4654309a448fd6d90f8ac8f
- Issue links:
    - https://github.com/metabase/metabase/issues/15483
    - https://github.com/metabase/metabase/issues/16315

**Implementation Plan**

This is most likely a 99% FE-only project. But depending on how we currently determine SSO users this could involve BE.

***Milestone 1 - Implement FE***
- [x] Hide first name, last name, and email field on the account page when logged in with SSO
- [ ] Fix JWT, SAML users can still edit their info https://github.com/metabase/metabase/pull/23476

**How to Test This Feature Out**
#### SSO user
- [ ] Setup SSO, currently only supposedly supports Google auth and LDAP.
- [ ] Login with SSO
- [ ] Go to the account page, should not see first name, last name, and email field anymore.

#### email + password user
- [ ] Login with email + password
- [ ] Go to the account page, should still see first name, last name, and email field and edit them.

### PR DESCRIPTION
**Pull request by WiNloSt**
Closes #23298

This PR alters the logic that checks if users are password/SSO by including `JWT` and `SAML` checks as well. See [this discussion on Slack](https://metaboat.slack.com/archives/C03E8MQJZBM/p1655843194210369) for more info

## Before this fix (JWT user)
image

## With this PR (JWT user)
image

### CODE REVIEW

**Review by adam-james-v — APPROVED** [review]
Nice stuff :) Looks good to me.

File: frontend/test/metabase/scenarios/onboarding/setup/user_settings.cy.spec.js [hunk 1]
```diff
@@ -214,4 +214,56 @@ describe("user > settings", () => {
...
+      cy.findByLabelText("First name").should("not.exist");
+      cy.findByLabelText("Last name").should("not.exist");
+      cy.findByLabelText("Email").should("not.exist");
+    });
+  });
+
+  describe("when user is authenticated via SAML", () => {
+    beforeEach(() => {
+      cy.server();
+      cy.route(
```
Comment (ranquild) [inline frontend/test/metabase/scenarios/onboarding/setup/user_settings.cy.spec.js:247]: `cy.route` is deprecated. Can we change it to [cy.intercept()](https://docs.cypress.io/api/commands/intercept)? There are many examples in our tests.

**Review by ranquild — APPROVED** [review]
Looks good, a comment about deprecated cypress methods

File: frontend/test/metabase/scenarios/onboarding/setup/user_settings.cy.spec.js (hunk 1 above)
Comment (WiNloSt) [reply frontend/test/metabase/scenarios/onboarding/setup/user_settings.cy.spec.js:247]: Good call. I just blindly copied from the adjacent tests. 😅

File: frontend/test/metabase/scenarios/onboarding/setup/user_settings.cy.spec.js [hunk 2]
```diff
@@ -64,8 +64,7 @@ describe("user > settings", () => {
   });
 
   it("should update the user without fetching memberships", () => {
-    cy.server();
-    cy.route("GET", "/api/permissions/membership").as("membership");
+    cy.intercept("GET", "/api/permissions/membership").as("membership");
```
Comment (WiNloSt) [inline frontend/test/metabase/scenarios/onboarding/setup/user_settings.cy.spec.js:67]: Refactor deprecated Cypress functions.

**Pull request merged**